test/acquisition/multi_objective/test_monte_carlo.py

```diff
@@ -1074,7 +1133,50 @@ def test_constrained_q_noisy_expected_hypervolume_improvement(self):
             )
             mm = MockModel(MockPosterior(samples=baseline_samples))
             X = torch.zeros(1, 1, **tkwargs)
-            # test zero slack
+            # test zero slack multiple constraints
+            for eta in (1e-1, 1e-2):
+                # set the MockPosterior to use samples over baseline points
+                mm._posterior._samples = baseline_samples
+                sampler = IIDNormalSampler(sample_shape=torch.Size([1]))
+                acqf = qNoisyExpectedHypervolumeImprovement(
+                    model=mm,
+                    ref_point=ref_point,
+                    X_baseline=X_baseline,
+                    sampler=sampler,
+                    constraints=[
+                        lambda Z: torch.zeros_like(Z[..., -1]),
+                        lambda Z: torch.zeros_like(Z[..., -1]),
+                    ],
+                    eta=eta,
+                    cache_root=False,
+                )
+                # set the MockPosterior to use samples over baseline points and new
+                # candidates
+                mm._posterior._samples = samples
+                res = acqf(X)
+                self.assertAlmostEqual(res.item(), 0.5 * 0.5 * 1.5, places=4)
+            # test zero slack multiple constraints, multiple etas
+            # set the MockPosterior to use samples over baseline points
+            mm._posterior._samples = baseline_samples
+            sampler = IIDNormalSampler(sample_shape=torch.Size([1]))
+            acqf = qNoisyExpectedHypervolumeImprovement(
+                model=mm,
+                ref_point=ref_point,
+                X_baseline=X_baseline,
+                sampler=sampler,
+                constraints=[
+                    lambda Z: torch.zeros_like(Z[..., -1]),
+                    lambda Z: torch.zeros_like(Z[..., -1]),
+                ],
+                eta=torch.tensor([1.0, 10.0]),
+                cache_root=False,
+            )
+            # set the MockPosterior to use samples over baseline points and new
+            # candidates
+            mm._posterior._samples = samples
+            res = acqf(X)
+            self.assertAlmostEqual(res.item(), 0.5 * 0.5 * 1.5, places=4)
```

Comment:
looks like this could also go in the for loop.

Reply:
Done.